tests/network/pod_networking.go

```diff
@@ -0,0 +1,101 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2020 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"fmt"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/tests"
+	cd "kubevirt.io/kubevirt/tests/containerdisk"
+)
+
+var _ = SIGDescribe("Pod Networking", func() {
+	var virtClient kubecli.KubevirtClient
+
+	BeforeEach(func() {
+		var err error
+		virtClient, err = kubecli.GetKubevirtClient()
+		tests.PanicOnError(err)
+	})
+
+	vmiWithDefaultBinding := func() *v1.VirtualMachineInstance {
+		vmi := tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
+		vmi.Spec.Domain.Devices.Interfaces = nil
+		vmi.Spec.Networks = nil
+		return vmi
+	}
+
+	vmiWithBridgeBinding := func() *v1.VirtualMachineInstance {
+		vmi := tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
+		vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultBridgeNetworkInterface()}
+		vmi.Spec.Networks = []v1.Network{*v1.DefaultPodNetwork()}
+		return vmi
+	}
+
+	vmiWithMasqueradeBinding := func() *v1.VirtualMachineInstance {
+		vmi := tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
+		vmi.Spec.Domain.Devices.Interfaces = []v1.Interface{*v1.DefaultMasqueradeNetworkInterface()}
+		vmi.Spec.Networks = []v1.Network{*v1.DefaultPodNetwork()}
+		return vmi
+	}
+
+	testStatusReporting := func(description string, vmiGenerator func() *v1.VirtualMachineInstance) {
```

Comment:
How about generalizing it?
Analyzing this parametrize shows that the Context and "Before" fixtures need it while the "It" does not (nor the "After" fixtures).

As ginkgo formatting encourage defining the tests in a chain of function closures, we can embrace it here as well and wrap it with a function call that in turn will call the (and register) the others.

Something like this:
```
ParametrizeClosure("Binding", values, f func(val param) {...})
```
In ParametrizeClosure then build the closure:
```
for _, val := range values {
    Context(description + val.description, func() {f(value)})
}
```

The values being a list of param types which enclose parametrize information (description, value, valueFunc, etc...)

What do you think?

Reply:
Ohh my... GitHub tricked me again and I looked at an older version :/

Reply:
So the current one is ok?